lib/vagrant/util/downloader.rb

```diff
@@ -203,9 +203,13 @@ def execute_curl(options, subprocess_options, &data_proc)
           @logger.warn("Downloader exit code: #{result.exit_code}")
           parts    = result.stderr.split(/\n*curl:\s+\(\d+\)\s*/, 2)
           parts[1] ||= ""
-          raise Errors::DownloaderError,
-            code: result.exit_code,
-            message: parts[1].chomp
+          if parts[1].include? "416"
+            # All good actually. 416 means there is no mory bytes to download
```

Comment:
Which is fine because then it would still fail on unpacking if file is it complete or corrupted. 

But for the case which I experienced in referenced issue it would actually make it work